lib/improver/tests/nbhood/circular_kernel/test_GeneratePercentilesFromACircularNeighbourhood.py

```diff
@@ -0,0 +1,647 @@
+
+    def test_single_point_almost_edge(self):
+        """Test behaviour for a non-zero grid cell quite near the edge."""
+        cube = set_up_cube(
+            zero_point_indices=[
+                (0, 0, 2, 2)],
+            num_grid_points=5)  # Just within range of the edge.
+        expected = np.array(
+            [[[[[0.66666667, 0.66666667, 0.66666667, 0.66666667, 0.66666667],
+                [0.66666667, 0.84444444, 0.93333333, 0.84444444, 0.66666667],
+                [0.66666667, 0.93333333, 0.66666667, 0.93333333, 0.66666667],
+                [0.66666667, 0.84444444, 0.93333333, 0.84444444, 0.66666667],
+                [0.66666667, 0.66666667, 0.66666667, 0.66666667, 0.66666667]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.]]]]])
+        percentiles = np.array([10, 50, 90])
+        radius = 6000.
+        result = (
+            GeneratePercentilesFromACircularNeighbourhood(
+                percentiles=percentiles).run(
+                    cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_adjacent_edge(self):
+        """Test behaviour for a single non-zero grid cell near the edge."""
+        cube = set_up_cube(
+            zero_point_indices=[(0, 0, 2, 1)],
+            num_grid_points=5)  # Range 3 goes over the edge.
+        expected = np.array(
+            [[[[[0.66666667, 0.66666667, 0.93333333, 0.93333333, 1.],
+                [0.66666667, 0.66666667, 1., 1., 1.],
+                [0.66666667, 0.66666667, 1., 1., 1.],
+                [0.66666667, 0.66666667, 1., 1., 1.],
+                [0.66666667, 0.66666667, 0.93333333, 0.93333333, 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.]]]]])
+        percentiles = np.array([10, 50, 90])
+        radius = 6000.
+        result = (
+            GeneratePercentilesFromACircularNeighbourhood(
+                percentiles=percentiles).run(
+                    cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_on_edge(self):
+        """Test behaviour for a non-zero grid cell on the edge.
+        Note that this behaviour is 'wrong' and is a result of
+        scipy.ndimage.correlate 'nearest' mode. We need to fix
+        this in the future."""
```

Comment:
Is this logged somewhere in a ticket, as I don't tend to read unit tests in my dull moments?

Reply:
Removed this and created #266.